chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -397,199 +399,207 @@
 node.set[:provisioner][:repositories] = Mash.new
 node.set[:provisioner][:available_oses] = Mash.new
 
-node[:provisioner][:supported_oses].each do |os,params|
-  web_path = "#{provisioner_web}/#{os}"
-  admin_web="#{web_path}/install"
-  crowbar_repo_web="#{web_path}/crowbar-extra"
-  os_dir="#{tftproot}/#{os}"
-  os_codename=node[:lsb][:codename]
-  role="#{os}_install"
-  missing_files = false
-  append = params["append"]
-  initrd = params["initrd"]
-  kernel = params["kernel"]
-  require_install_dir = params["require_install_dir"].nil? ? true : params["require_install_dir"]
-
-  if require_install_dir
-    # Don't bother for OSes that are not actually present on the provisioner node.
-    next unless File.directory? os_dir and File.directory? "#{os_dir}/install"
-  end
-
-  # Index known barclamp repositories for this OS
-  node[:provisioner][:repositories][os] ||= Mash.new
-  if File.exists? "#{os_dir}/crowbar-extra" and File.directory? "#{os_dir}/crowbar-extra"
-    Dir.foreach("#{os_dir}/crowbar-extra") do |f|
-      next unless File.symlink? "#{os_dir}/crowbar-extra/#{f}"
-      node[:provisioner][:repositories][os][f] ||= Hash.new
-      case
-      when os =~ /(ubuntu|debian)/
-        bin="deb http://#{admin_ip}:#{web_port}/#{os}/crowbar-extra/#{f} /"
-        src="deb-src http://#{admin_ip}:#{web_port}/#{os}/crowbar-extra/#{f} /"
-        node.set[:provisioner][:repositories][os][f][bin] = true if
-          File.exists? "#{os_dir}/crowbar-extra/#{f}/Packages.gz"
-        node.set[:provisioner][:repositories][os][f][src] = true if
-          File.exists? "#{os_dir}/crowbar-extra/#{f}/Sources.gz"
-      when os =~ /(redhat|centos|suse)/
-        bin="baseurl=http://#{admin_ip}:#{web_port}/#{os}/crowbar-extra/#{f}"
-        node.set[:provisioner][:repositories][os][f][bin] = true
-        else
-          raise ::RangeError.new("Cannot handle repos for #{os}")
-        end
+node[:provisioner][:supported_oses].each do |os, arches|
+  arches.each do |arch, params|
+    web_path = "#{provisioner_web}/#{os}/#{arch}"
+    admin_web="#{web_path}/install"
+    crowbar_repo_web="#{web_path}/crowbar-extra"
+    os_dir="#{tftproot}/#{os}/#{arch}"
```

Comment:
Style/SpaceAroundOperators: Surrounding space missing for operator `=`. (https://github.com/bbatsov/ruby-style-guide#spaces-operators)